contracts/v2/self-service/SelfServiceEditionCurationV4.sol

```diff
@@ -0,0 +1,298 @@
+pragma solidity 0.4.24;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "../interfaces/IKODAV2SelfServiceEditionCuration.sol";
+import "../interfaces/IKODAAuction.sol";
+import "../interfaces/ISelfServiceAccessControls.sol";
+import "../interfaces/ISelfServiceFrequencyControls.sol";
+
+// One invocation per time-period
+contract SelfServiceEditionCurationV4 is Ownable, Pausable {
+  using SafeMath for uint256;
+
+  event SelfServiceEditionCreated(
+    uint256 indexed _editionNumber,
+    address indexed _creator,
+    uint256 _priceInWei,
+    uint256 _totalAvailable,
+    bool _enableAuction
+  );
+
+  // Calling address
+  IKODAV2SelfServiceEditionCuration public kodaV2;
+  IKODAAuction public auction;
+  ISelfServiceAccessControls public accessControls;
+  ISelfServiceFrequencyControls public frequencyControls;
+
+  // Default KO commission
+  uint256 public koCommission = 15;
+
+  // Config which enforces editions to not be over this size
+  uint256 public maxEditionSize = 100;
+
+  // Config the minimum price per edition
+  uint256 public minPricePerEdition = 0.01 ether;
+
+  /**
+   * @dev Construct a new instance of the contract
+   */
+  constructor(
+    IKODAV2SelfServiceEditionCuration _kodaV2,
+    IKODAAuction _auction,
+    ISelfServiceAccessControls _accessControls,
+    ISelfServiceFrequencyControls _frequencyControls
+  ) public {
+    kodaV2 = _kodaV2;
+    auction = _auction;
+    accessControls = _accessControls;
+    frequencyControls = _frequencyControls;
+  }
+
+  /**
+   * @dev Called by artists, create new edition on the KODA platform
+   */
+  function createEdition(
+    bool _enableAuction,
+    address _optionalSplitAddress,
```

Comment:
Let's face it - they are going to want to create multi-splits between x artists. Worth addressing that now?

Reply:
Agree, not sure how we do that easily though? Ideas?